src/charm.py

```diff
@@ -64,11 +66,19 @@ def _on_config_changed(self, event: ops.HookEvent) -> None:
             event.defer()
             self.unit.status = ops.WaitingStatus("Waiting for pebble")
             return
-        self.model.unit.status = ops.MaintenanceStatus("Configuring Synapse")
         try:
+            self._charm_state = CharmState.from_charm(charm=self)
+            self._synapse = Synapse(charm_state=self._charm_state)
```

Comment:
Similarly to the CharmState, this service is already instanciated in the constructor, so I think it is not needed here

Reply:
I'll close this one in favor of https://github.com/canonical/synapse-operator/pull/8#discussion_r1245192606 so we can discuss there.